task2/task2.2.js

```diff
@@ -11,111 +21,125 @@ app.use(function(req, _res, next) {
-        }
-
-        if (this.users.some(u => u.login === login)) {
-            throw new Error(`Cannot create - user with the same login ${login} already exists`);
-        }
-        const newUser = new User(id, login, password, age);
-        this.users.push(newUser);
-        return newUser.id;
+    async create({id, login, password, age}) {
+        const user = await User.create({ id, login, password, age });
+        return user.id;
     }
 
-    update({id, login, password, age}) {
-        const user = this.find(id);
-        if (!user || user.isDeleted === true) {
-            throw new Error(`Cannot update - user id ${id} does not exists or removed`);
-        }
-
-        if (login) user.login = login;
-        if (password) user.password = password;
-        if (age) user.age = age;
+    async update({id, login, password, age}) {
+        await User.update({login, password, age}, {
+            where: {id}
+        });
     }
 
-    delete(id, soft = true) {
-        const user = this.find(id);
-        if (!user) {
-            throw new Error(`Cannot remove - user id ${id} does not exists`);
-        }
+    async delete(id, soft = true) {
         if (soft) {
-            user.isDeleted = true;
+            await User.update({isDeleted: true}, {
+                where: {id}
+            });
         } else {
-            this.users = this.users.filter(u => u.id !== id);
+            await User.destroy({
+                where: {id}
+            });
         }
     }
 
-    getAutoSuggestUsers(loginSubstring, limit = 0) {
-        const retUsers = this.getAll()
-            .filter(u => u.login && u.login.includes(loginSubstring))
-            .sort((u1, u2) => u1.login > u2.login ? 1 :
-                            u1.login < u2.login ? -1 : 0);
-        return limit > 0 ? retUsers.slice(0, limit) : retUsers;
+    async getAutoSuggestUsers(loginSubstring, limit) {
+        if (!(limit > 0)) { // validation for non-number
+            limit = undefined;
+        }
+        return User.findAll({
+            order: sequelize.col('login'),
+            limit,
+            attributes: {exclude: ['isDeleted']},
+            where: {
+                login: {
+                    [Op.like]: `%${loginSubstring}%`
+                },
+                isDeleted: {[Op.not]: true}
+            }
+        });
     }
 
-    getAll() {
-        console.log(this.users); // for debug 
-        return this.users.reduce((prev, curr) => {
-            if (curr.isDeleted !== true) {
-                const retUser = { ...curr };
-                delete retUser.isDeleted;
-                prev.push(retUser);
+    async getAll() {
+        return User.findAll({
+            attributes: {exclude: ['isDeleted']},
+            where: {
+                isDeleted: {[Op.not]: true}
             }
-            return prev;
-        }, []);
+        });
     }
 }
 
 const users_db = new Users();

```

Comment:
user_db is not really interacting with DB. It's just naming, but it might confuse other developers